src/shogun/neuralnets/NeuralNetwork.cpp

```diff
@@ -229,6 +238,15 @@ CDenseFeatures< float64_t >* CNeuralNetwork::transform(
 
 bool CNeuralNetwork::train_machine(CFeatures* data)
 {
+	if (!m_is_connected)
+	{
+		quick_connect();
+	}
+	if (!m_is_initialized)
```

Comment:
this is cleaner since it will avoid multiple calls to initialize_neural_network

Reply:
yep!

Reply:
although .... 
why would there be multiple calls? Due to multiple "train" calls?

Reply:
also would need to make sure that these things are set to false if the network architecture is changed.
So actually, it might be that *always* calling these might be a better idea. The cost is neglectible.
Thoughts?